### ISSUE TITLE
Fluentd Docker Log Driver with Docker 18.x does not work with 1.5

### ISSUE THREAD

**Issue by PettitWesley** [fluent/fluent-bit#2355]
## Bug Report

This does not seem to affect 19.x Docker versions. It also only affects Fluent Bit 1.5; not any older versions. I can confirm that this bug affects the following versions of Docker community addition:
- 18.09.0
- 18.09.9
- 18.06.1

I tested those Docker versions on Amazon Linux and Ubuntu.

My suspicion is that all 18.x versions and lower are affected.

This message will be logged repeatedly:
```
[2020/07/14 04:06:48] [debug] [input:forward:forward.0] invalid options field
```

The log messages from Docker are discarded by in_forward and not sent to the output.

I think this commit added the bug: https://github.com/fluent/fluent-bit/commit/f838308dc8b38c69353cc1e48e125965f414d220

I spent some time playing with the code and this line is the one where the message is generated: https://github.com/fluent/fluent-bit/blob/master/plugins/in_forward/fw_prot.c#L399

Something is going wrong there.

**To Reproduce**
Any simple config for in_forward works:

```
[SERVICE]
          Log_Level debug
 
[INPUT]
    Name forward
    Listen 0.0.0.0
    Port 24224
 
[OUTPUT]
        Name stdout
        Match *
```

Then, run any docker image that produces logs:

```
docker run -it --log-driver fluentd app
```

**Expected behavior**
With Fluent Bit <1.5, this issue does not appear.

**Comment by PettitWesley** [comment]
I am concerned that this issue will hurt AWS customers; the Docker 18.x series is still widely used. AWS for Fluent Bit will release and consume Fluent Bit 1.5, but we will not include the commit that introduced this bug.

**Comment by edsiper** [comment]
@PettitWesley thanks for reporting the problem.

Troubleshooting I found Docker 18.x fluentd driver sends a ```null``` third parameter in the protocol to represent a missing options. 

I've pushed the following fix in GIT master: d9ace9f36c07161abcf4560688619af6b68b91be

**Comment by edsiper** [comment]
Fix ported to v1.5 branch for v1.5.1 release. closing.
